code/modules/atmospherics/environmental/LINDA_turf_tile.dm

```diff
@@ -175,64 +174,48 @@
 	for(var/t in adjacent_turfs)
 		var/turf/open/enemy_tile = t
 
-		if(fire_count > enemy_tile.current_cycle)
-			enemy_tile.archive()
-
-		/******************* GROUP HANDLING START *****************************************************************/
-
-			var/should_share_air = FALSE
-			var/datum/gas_mixture/enemy_air = enemy_tile.air
-			if(enemy_tile.excited)
-				//cache for sanic speed
-				var/datum/excited_group/enemy_excited_group = enemy_tile.excited_group
-				if(our_excited_group)
-					if(enemy_excited_group)
-						if(our_excited_group != enemy_excited_group)
-							//combine groups (this also handles updating the excited_group var of all involved turfs)
-							our_excited_group.merge_groups(enemy_excited_group)
-							our_excited_group = excited_group //update our cache
-						should_share_air = TRUE
-					else
-						if((recently_active == 1 && enemy_tile.recently_active == 1) || our_air.compare(enemy_air))
-							our_excited_group.add_turf(enemy_tile) //add enemy to our group
-							should_share_air = TRUE
+		if(fire_count <= enemy_tile.current_cycle)
+			continue
+		enemy_tile.archive()
+
+	/******************* GROUP HANDLING START *****************************************************************/
+
+		var/should_share_air = FALSE
+		var/datum/gas_mixture/enemy_air = enemy_tile.air
+
+		//cache for sanic speed
+		var/datum/excited_group/enemy_excited_group = enemy_tile.excited_group
+
+		if(our_excited_group && enemy_excited_group)
+			if(our_excited_group != enemy_excited_group)
+				//combine groups (this also handles updating the excited_group var of all involved turfs)
+				our_excited_group.merge_groups(enemy_excited_group)
+				our_excited_group = excited_group //update our cache
+			should_share_air = TRUE
+
+		else if(our_air.compare(enemy_air))
+			if(!enemy_tile.excited)
+				SSair.add_to_active(enemy_tile)
+			var/datum/excited_group/EG = our_excited_group || enemy_excited_group || new
```

Comment:
I'm slightly worried this line will always create a new excited group, even if it doesn't use it. Did you test that doesn't happen?

Reply:
it only happens if the first two are false, ```||``` is a short-circuiting operator.

Reply:
Yeah, that's how it should work, but BYOND makes me paranoid

Reply:
It's how it should and <i>does</i> work.

Reply:
Short circuiting is one of the few well documented and trustworthy behaviors we have in BYOND. haven't you ever written `if(A && A.whatever)` to prevent runtimes when A is null? that wouldn't work if we didn't short circuit properly

Reply:
Protip, if you do a if (0 && something...), byond will literally not even compile the shit after the 0 because it will never trigger, same for if (1 || blah)


````
proc/blah()
  var/blah = args
  if (0 && blah)
````

````
Warning: blah: var defined but not used.
````